### PR TITLE
v1.2.2

### LINKED ISSUE #250: Having more than `role` and `description` in portrait component
### Checks

- [X] I have searched **[the docs](https://greene-lab.gitbook.io/lab-website-template-docs)**, [existing issues](https://github.com/greenelab/lab-website-template/issues), and [existing discussions](https://github.com/greenelab/lab-website-template/discussions) for answers first.

### Link to your website repo

https://github.com/sbbecongresso/sbbecongresso.github.io

### Version of Lab Website Template you are using

1.2.1

### Description

Hello. 

Regarding the members, I am aware that the description will override the role. However, I would like to know if there would be a way to keep two or more entries like NAME, ROLE & AFFILIATION for instance:

**George Pacheco**
Postdoctoral Researcher
University of Oslo

I did try to change the code to achieve such a result to no avail. I think it is somewhere in here:

```
        {% if member.role or member.description %}
          <span class="portrait-role">
            {% assign type = site.data.types[member.role] %}
            {% include icon.html icon=type.icon %}
            <span>{{ member.description | default: member.role type.description member.affiliation }}</span>
```

Many thanks in advance, George.

### LINKED ISSUE #256: Tags component doesn't de-dupe case-insensitively
However the [search functionality](https://greene-lab.gitbook.io/lab-website-template-docs/basics/components/search) de-duplicates, the tags component should de-dupe in the same way, since the main purpose of the tags is to click on them and search the page.

### LINKED ISSUE #258: Investigate which id/date/etc fields to use from ORCID API in `orcid.py`
### Checks

- [X] I have searched **[the docs](https://greene-lab.gitbook.io/lab-website-template-docs)**, [existing issues](https://github.com/greenelab/lab-website-template/issues), and [existing discussions](https://github.com/greenelab/lab-website-template/discussions) for answers first.

### Link to your website repo

https://github.com/andrewsu/sulab.org

### Version of Lab Website Template you are using

1.2.1

### Description

**TLDR:  I think `orcid.py` should be updated to get the `external-id` from the first `work-summary` entry, not from the first doi in `external-ids.external-id`.**

## Observed/desired behavior

On https://andrewsu.github.io/sulab.org/members/andrew-su.html, I see the following citation

image

The citation points to [the biorxiv preprint](https://doi.org/gtrg2h) (note that CSHL is the publisher/journal). However, in [my ORCID profile](https://orcid.org/0000-0002-9859-4104), the preprint is grouped with [the published article at Bioinformatics](https://doi.org/10.1093/bioinformatics/btac205), and it is that version of the article that I've listed as the "Preferred source".

image

Therefore, I would expect that the citation on my profile page would list "Bioinformatics" as the journal/publisher and the link would go to the journal website, not biorxiv.

## My diagnosis

When `orcid.py` retrieves my ORCID profile from https://pub.orcid.org/v3.0/0000-0002-9859-4104/works, this is the XML for the article above:

<details>
<summary>XML snippet for article above</summary>

```
    <activities:group>
        <common:last-modified-date>2024-04-21T02:40:56.931Z</common:last-modified-date>
        <common:external-ids>
            <common:external-id>
                <common:external-id-type>doi</common:external-id-type>
                <common:external-id-value>10.1101/2021.04.15.440028</common:external-id-value>
                <common:external-id-normalized transient="true">10.1101/2021.04.15.440028</common:external-id-normalized>
                <common:external-id-url>https://doi.org/10.1101/2021.04.15.440028</common:external-id-url>
                <common:external-id-relationship>self</common:external-id-relationship>
            </common:external-id>
            <common:external-id>
                <common:external-id-type>doi</common:external-id-type>
                <common:external-id-value>10.1093/bioinformatics/btac205</common:external-id-value>
                <common:external-id-normalized transient="true">10.1093/bioinformatics/btac205</common:external-id-normalized>
                <common:external-id-url>https://doi.org/10.1093/bioinformatics/btac205</common:external-id-url>
                <common:external-id-relationship>self</common:external-id-relationship>
            </common:external-id>
        </common:external-ids>
        <work:work-summary put-code="111796297" path="/0000-0002-9859-4104/work/111796297" visibility="public" display-index="3">
            <common:created-date>2022-04-21T09:36:24.476Z</common:created-date>
            <common:last-modified-date>2024-04-21T02:40:56.931Z</common:last-modified-date>
            <common:source>
                <common:source-client-id>
                    <common:uri>https://orcid.org/client/0000-0001-9884-1913</common:uri>
                    <common:path>0000-0001-9884-1913</common:path>
                    <common:host>orcid.org</common:host>
                </common:source-client-id>
                <common:source-name>Crossref</common:source-name>
            </common:source>
            <work:title>
                <common:title>Design and application of a knowledge network for automatic prioritization of drug mechanisms</common:title>
            </work:title>
            <common:external-ids>
                <common:external-id>
                    <common:external-id-type>doi</common:external-id-type>
                    <common:external-id-value>10.1093/bioinformatics/btac205</common:external-id-value>
                    <common:external-id-normalized transient="true">10.1093/bioinformatics/btac205</common:external-id-normalized>
                    <common:external-id-url>https://doi.org/10.1093/bioinformatics/btac205</common:external-id-url>
                    <common:external-id-relationship>self</common:external-id-relationship>
                </common:external-id>
            </common:external-ids>
            <common:url>https://doi.org/10.1093/bioinformatics/btac205</common:url>
            <work:type>journal-article</work:type>
            <common:publication-date>
                <common:year>2022</common:year>
                <common:month>05</common:month>
                <common:day>13</common:day>
            </common:publication-date>
            <work:journal-title>Bioinformatics</work:journal-title>
        </work:work-summary>
        <work:work-summary put-code="92294640" path="/0000-0002-9859-4104/work/92294640" visibility="public" display-index="0">
            <common:created-date>2021-04-16T08:27:32.678Z</common:created-date>
            <common:last-modified-date>2022-06-01T04:14:25.366Z</common:last-modified-date>
            <common:source>
                <common:source-client-id>
                    <common:uri>https://orcid.org/client/0000-0001-9884-1913</common:uri>
                    <common:path>0000-0001-9884-1913</common:path>
                    <common:host>orcid.org</common:host>
                </common:source-client-id>
                <common:source-name>Crossref</common:source-name>
            </common:source>
            <work:title>
                <common:title>Design and application of a knowledge network for automatic prioritization of drug mechanisms</common:title>
            </work:title>
            <common:external-ids>
                <common:external-id>
                    <common:external-id-type>doi</common:external-id-type>
                    <common:external-id-value>10.1101/2021.04.15.440028</common:external-id-value>
                    <common:external-id-normalized transient="true">10.1101/2021.04.15.440028</common:external-id-normalized>
                    <common:external-id-url>https://doi.org/10.1101/2021.04.15.440028</common:external-id-url>
                    <common:external-id-relationship>self</common:external-id-relationship>
                </common:external-id>
            </common:external-ids>
            <common:url>https://doi.org/10.1101/2021.04.15.440028</common:url>
            <work:type>other</work:type>
            <common:publication-date>
                <common:year>2021</common:year>
                <common:month>04</common:month>
                <common:day>15</common:day>
            </common:publication-date>
        </work:work-summary>
        <work:work-summary put-code="158033604" path="/0000-0002-9859-4104/work/158033604" visibility="public" display-index="-1">
            <common:created-date>2024-04-19T22:48:31.846Z</common:created-date>
            <common:last-modified-date>2024-04-19T22:48:31.846Z</common:last-modified-date>
            <common:source>
                <common:source-orcid>
                    <common:uri>https://orcid.org/0000-0002-9859-4104</common:uri>
                    <common:path>0000-0002-9859-4104</common:path>
                    <common:host>orcid.org</common:host>
                </common:source-orcid>
                <common:source-name>Andrew I. Su</common:source-name>
            </common:source>
            <work:title>
                <common:title>Design and application of a knowledge network for automatic prioritization of drug mechanisms</common:title>
            </work:title>
            <common:external-ids>
                <common:external-id>
                    <common:external-id-type>doi</common:external-id-type>
                    <common:external-id-value>10.1101/2021.04.15.440028</common:external-id-value>
                    <common:external-id-normalized transient="true">10.1101/2021.04.15.440028</common:external-id-normalized>
                    <common:external-id-url>https://doi.org/10.1101/2021.04.15.440028</common:external-id-url>
                    <common:external-id-relationship>self</common:external-id-relationship>
                </common:external-id>
                <common:external-id>
                    <common:external-id-type>doi</common:external-id-type>
                    <common:external-id-value>10.1093/bioinformatics/btac205</common:external-id-value>
                    <common:external-id-normalized transient="true">10.1093/bioinformatics/btac205</common:external-id-normalized>
                    <common:external-id-url>https://doi.org/10.1093/bioinformatics/btac205</common:external-id-url>
                    <common:external-id-relationship>self</common:external-id-relationship>
                </common:external-id>
            </common:external-ids>
            <common:url>https://doi.org/10.1101/2021.04.15.440028</common:url>
            <work:type>other</work:type>
            <common:publication-date>
                <common:year>2021</common:year>
                <common:month>04</common:month>
                <common:day>15</common:day>
            </common:publication-date>
        </work:work-summary>
    </activities:group>
```
</details>

Currently, the code gets the list of IDs from `external-ids.external-id` section, and I believe it takes the first entry with a `external-id-type` of `doi` -- in this case, `10.1101/2021.04.15.440028`. Unfortunately, this section does not look like there is any indication of the "preferred source". However, it appears that in the `work-summary` entries are listed so that the "preferred source" is listed first -- `10.1093/bioinformatics/btac205` is first.

I checked this behavior against another entry in my profile where the biorxiv article is listed as primary. See details below.

<details>
<summary>ORCID screenshot and XML snippet where biorxiv is listed as "preferred source"</summary>

image

```
    <activities:group>
        <common:last-modified-date>2024-04-19T22:48:31.838Z</common:last-modified-date>
        <common:external-ids>
            <common:external-id>
                <common:external-id-type>doi</common:external-id-type>
                <common:external-id-value>10.1101/367599</common:external-id-value>
                <common:external-id-normalized transient="true">10.1101/367599</common:external-id-normalized>
                <common:external-id-url>https://doi.org/10.1101/367599</common:external-id-url>
                <common:external-id-relationship>self</common:external-id-relationship>
            </common:external-id>
            <common:external-id>
                <common:external-id-type>doi</common:external-id-type>
                <common:external-id-value>10.1021/acs.jproteome.8b00166</common:external-id-value>
                <common:external-id-normalized transient="true">10.1021/acs.jproteome.8b00166</common:external-id-normalized>
                <common:external-id-relationship>self</common:external-id-relationship>
            </common:external-id>
            <common:external-id>
                <common:external-id-type>eid</common:external-id-type>
                <common:external-id-value>2-s2.0-85050359010</common:external-id-value>
                <common:external-id-normalized transient="true">2-s2.0-85050359010</common:external-id-normalized>
                <common:external-id-relationship>self</common:external-id-relationship>
            </common:external-id>
        </common:external-ids>
        <work:work-summary put-code="46481891" path="/0000-0002-9859-4104/work/46481891" visibility="public" display-index="0">
            <common:created-date>2018-07-11T19:50:03.575Z</common:created-date>
            <common:last-modified-date>2022-05-27T03:15:05.789Z</common:last-modified-date>
            <common:source>
                <common:source-client-id>
                    <common:uri>https://orcid.org/client/0000-0001-9884-1913</common:uri>
                    <common:path>0000-0001-9884-1913</common:path>
                    <common:host>orcid.org</common:host>
                </common:source-client-id>
                <common:source-name>Crossref</common:source-name>
            </common:source>
            <work:title>
                <common:title>Triflic acid treatment enables LC-MS/MS analysis of insoluble bacterial biomass</common:title>
            </work:title>
            <common:external-ids>
                <common:external-id>
                    <common:external-id-type>doi</common:external-id-type>
                    <common:external-id-value>10.1101/367599</common:external-id-value>
                    <common:external-id-normalized transient="true">10.1101/367599</common:external-id-normalized>
                    <common:external-id-url>https://doi.org/10.1101/367599</common:external-id-url>
                    <common:external-id-relationship>self</common:external-id-relationship>
                </common:external-id>
            </common:external-ids>
            <common:url>https://doi.org/10.1101/367599</common:url>
            <work:type>other</work:type>
            <common:publication-date>
                <common:year>2018</common:year>
                <common:month>07</common:month>
                <common:day>11</common:day>
            </common:publication-date>
        </work:work-summary>
        <work:work-summary put-code="54674396" path="/0000-0002-9859-4104/work/54674396" visibility="public" display-index="0">
            <common:created-date>2019-02-27T20:02:31.774Z</common:created-date>
            <common:last-modified-date>2022-05-27T21:53:54.347Z</common:last-modified-date>
            <common:source>
                <common:source-client-id>
                    <common:uri>https://orcid.org/client/0000-0002-5982-8983</common:uri>
                    <common:path>0000-0002-5982-8983</common:path>
                    <common:host>orcid.org</common:host>
                </common:source-client-id>
                <common:source-name>Scopus - Elsevier</common:source-name>
                <common:assertion-origin-orcid>
                    <common:uri>https://orcid.org/0000-0002-9859-4104</common:uri>
                    <common:path>0000-0002-9859-4104</common:path>
                    <common:host>orcid.org</common:host>
                </common:assertion-origin-orcid>
                <common:assertion-origin-name>Andrew I. Su</common:assertion-origin-name>
            </common:source>
            <work:title>
                <common:title>Triflic Acid Treatment Enables LC-MS/MS Analysis of Insoluble Bacterial Biomass</common:title>
            </work:title>
            <common:external-ids>
                <common:external-id>
                    <common:external-id-type>doi</common:external-id-type>
                    <common:external-id-value>10.1021/acs.jproteome.8b00166</common:external-id-value>
                    <common:external-id-normalized transient="true">10.1021/acs.jproteome.8b00166</common:external-id-normalized>
                    <common:external-id-relationship>self</common:external-id-relationship>
                </common:external-id>
                <common:external-id>
                    <common:external-id-type>eid</common:external-id-type>
                    <common:external-id-value>2-s2.0-85050359010</common:external-id-value>
                    <common:external-id-normalized transient="true">2-s2.0-85050359010</common:external-id-normalized>
                    <common:external-id-relationship>self</common:external-id-relationship>
                </common:external-id>
            </common:external-ids>
            <common:url>http://www.scopus.com/inward/record.url?eid=2-s2.0-85050359010&amp;partnerID=MN8TOARS</common:url>
            <work:type>journal-article</work:type>
            <common:publication-date>
                <common:year>2018</common:year>
            </common:publication-date>
            <work:journal-title>Journal of Proteome Research</work:journal-title>
        </work:work-summary>
        <work:work-summary put-code="158033601" path="/0000-0002-9859-4104/work/158033601" visibility="public" display-index="-1">
            <common:created-date>2024-04-19T22:48:31.838Z</common:created-date>
            <common:last-modified-date>2024-04-19T22:48:31.838Z</common:last-modified-date>
            <common:source>
                <common:source-orcid>
                    <common:uri>https://orcid.org/0000-0002-9859-4104</common:uri>
                    <common:path>0000-0002-9859-4104</common:path>
                    <common:host>orcid.org</common:host>
                </common:source-orcid>
                <common:source-name>Andrew I. Su</common:source-name>
            </common:source>
            <work:title>
                <common:title>Triflic acid treatment enables LC-MS/MS analysis of insoluble bacterial biomass</common:title>
            </work:title>
            <common:external-ids>
                <common:external-id>
                    <common:external-id-type>doi</common:external-id-type>
                    <common:external-id-value>10.1101/367599</common:external-id-value>
                    <common:external-id-normalized transient="true">10.1101/367599</common:external-id-normalized>
                    <common:external-id-url>https://doi.org/10.1101/367599</common:external-id-url>
                    <common:external-id-relationship>self</common:external-id-relationship>
                </common:external-id>
                <common:external-id>
                    <common:external-id-type>doi</common:external-id-type>
                    <common:external-id-value>10.1021/acs.jproteome.8b00166</common:external-id-value>
                    <common:external-id-normalized transient="true">10.1021/acs.jproteome.8b00166</common:external-id-normalized>
                    <common:external-id-relationship>self</common:external-id-relationship>
                </common:external-id>
                <common:external-id>
                    <common:external-id-type>eid</common:external-id-type>
                    <common:external-id-value>2-s2.0-85050359010</common:external-id-value>
                    <common:external-id-normalized transient="true">2-s2.0-85050359010</common:external-id-normalized>
                    <common:external-id-relationship>self</common:external-id-relationship>
                </common:external-id>
            </common:external-ids>
            <common:url>https://doi.org/10.1101/367599</common:url>
            <work:type>other</work:type>
            <common:publication-date>
                <common:year>2018</common:year>
                <common:month>07</common:month>
                <common:day>11</common:day>
            </common:publication-date>
        </work:work-summary>
    </activities:group>
```

Note that `10.1101/367599` is listed in the first `work-summary` section.
</details>

Based on all this, I think `orcid.py` should be updated to get the `external-id` from the first `work-summary` entry, not from the first doi in `external-ids.external-id`.

### LINKED ISSUE #260: Use `created-date` instead of `last-modified-date` in `orcid.py`
### Checks

- [X] I have searched **[the docs](https://greene-lab.gitbook.io/lab-website-template-docs)**, [existing issues](https://github.com/greenelab/lab-website-template/issues), and [existing discussions](https://github.com/greenelab/lab-website-template/discussions) for answers first.

### Link to your website repo

https://github.com/andrewsu/sulab.org

### Version of Lab Website Template you are using

1.2.1

### Description

Out of the 156 citations currently in the `citations.yaml`, all but three of the IDs use the `doi` prefix.  The ones that don't are: `pmc:PMC3782358`, `pmid:11606367`, and `pmid:11507037`. For all three of these, the dates retrieved are incorrect. Taking `pmid:11606367` as an example, https://pubmed.ncbi.nlm.nih.gov/11606367 and https://aacrjournals.org/cancerres/article/61/20/7388/508137/Molecular-Classification-of-Human-Carcinomas-by show that this was published in 2001, but the entry in `citations.yaml` says the publication date is in 2022:

```
- id: pmid:11606367
  title: Molecular classification of human carcinomas by use of gene expression signatures.
  authors:
  - A I Su
  - J B Welsh
  - L M Sapinoso
  - S G Kern
  - P Dimitrov
  - H Lapp
  - P G Schultz
  - S M Powell
  - C A Moskaluk
  - H F Frierson
  - G M Hampton
  publisher: Cancer research
  date: '2022-05-23'
  link: http://europepmc.org/abstract/med/11606367
  orcid: 0000-0002-9859-4104
  member: andrew-su
  plugin: orcid.py
  file: orcid.yaml
```

It looks like orcid.py is pulling that from my ORCID profile via the API.  Here is the relevant snippet:

```
<activities:group>
	<common:last-modified-date>2022-05-23T23:38:16.840Z</common:last-modified-date>
	<common:external-ids>
		<common:external-id>
			<common:external-id-type>pmid</common:external-id-type>
			<common:external-id-value>11606367</common:external-id-value>
			<common:external-id-normalized transient="true">11606367</common:external-id-normalized>
			<common:external-id-url/>
			<common:external-id-relationship>self</common:external-id-relationship>
		</common:external-id>
	</common:external-ids>
	<work:work-summary put-code="13492064" path="/0000-0002-9859-4104/work/13492064" visibility="public" display-index="0">
		<common:created-date>2014-08-28T23:46:45.243Z</common:created-date>
		<common:last-modified-date>2022-05-23T23:38:16.840Z</common:last-modified-date>
		<common:source>
			<common:source-client-id>
				<common:uri>https://orcid.org/client/0000-0002-9157-3431</common:uri>
				<common:path>0000-0002-9157-3431</common:path>
				<common:host>orcid.org</common:host>
			</common:source-client-id>
			<common:source-name>Europe PubMed Central</common:source-name>
			<common:assertion-origin-orcid>
				<common:uri>https://orcid.org/0000-0002-9859-4104</common:uri>
				<common:path>0000-0002-9859-4104</common:path>
				<common:host>orcid.org</common:host>
			</common:assertion-origin-orcid>
			<common:assertion-origin-name>Andrew I. Su</common:assertion-origin-name>
		</common:source>
		<work:title>
			<common:title>Molecular classification of human carcinomas by use of gene expression signatures.</common:title>
		</work:title>
		<common:external-ids>
			<common:external-id>
				<common:external-id-type>pmid</common:external-id-type>
				<common:external-id-value>11606367</common:external-id-value>
				<common:external-id-normalized transient="true">11606367</common:external-id-normalized>
				<common:external-id-url/>
				<common:external-id-relationship>self</common:external-id-relationship>
			</common:external-id>
		</common:external-ids>
		<common:url>http://europepmc.org/abstract/med/11606367</common:url>
		<work:type>journal-article</work:type>
		<common:publication-date>
			<common:year>2001</common:year>
			<common:month>10</common:month>
		</common:publication-date>
	</work:work-summary>
</activities:group>
```

And it looks like `orcid.py` is pulling from `last-modified-date` instead of `publication-date`, and that's the cause of the error in these particular cases.  

Since it's just three entries, I'll override the dates for these entries in my `sources.yaml`. But just making the issue so you can assess whether a broader change is warranted...

### PR DESCRIPTION
**Pull request by vincerubinetti**
Closes #250 
Closes #256 
Closes #260 
Closes #258

- adds `affiliation` to member portrait component
- simplify portrait component code/css
- make tag de-dupe behavior the same as search de-dupe. normalize to lower-kebab-case.
- expand list of manubot-supported id types for falling back to orcid api details
- change order and type of preferred ids from orcid

### CODE REVIEW

**Comment by vincerubinetti** [comment]
@falquaddoomi Can you take a look at this? The main changes of note are in `orcid.py`, and the more possibly questionable changes involve issue https://github.com/greenelab/lab-website-template/issues/258. Summary: not sure which ORCID API fields to use. It returns a lot of confusing and duplicate fields, their documentation is insufficient, and there's basically no one I can contact to clarify. We have one data point (Andrew Su) where these changes match their expectations, and one data point (Casey Greene) where these changes don't produce any different results.

**Comment by falquaddoomi** [comment]
Hey @vincerubinetti; agreed, the ORCID API docs leave a lot to be desired.

Regarding issue #258, I think I was able to replicate the behavior Andrew reported with a publication in my ORCID account that has three sources: changing the preferred source changed the order of the elements in `work-summary` for the publication's group, as well as changing the `display-index` for that source. There's a bit from the ORCID API tutorial on the display index (https://info.orcid.org/documentation/api-tutorials/api-tutorial-read-data-on-a-record/, in the "How are items grouped together in an ORCID record?" expandable section):
> Our API provides support for this in the XSD. Each item has a display index attribute which indicates its rank within its group. The highest display index is the preferred item selected by the researcher, items added via the API which have not been ranked by the researcher have a display index of 0. The display index also determines the work order when reading the ORCID record.

I don't know if that's enough to resolve that issue, but I hope it helps.

**Comment by vincerubinetti** [comment]
That's great, thanks for testing that out. None of my ORCID entries had multiple work summaries to test that with. With that extra data point, I feel confident enough that the api will do what it says (user-specified preferred sources become the highest `display-index`, and api returns in `display-index` order) and that this is the right decision.

File: _cite/plugins/orcid.py [hunk 1]
```diff
@@ -35,25 +36,34 @@ def query(_id):
...
-        # prefer doi id type, or fallback to first id
+        # prefer particular "relationship" type, or fallback to first id
         _id = next(
-            (id for id in ids if get_safe(id, "external-id-type", "") == "doi"),
-            ids[0] if len(ids) > 0 else {},
+            (
+                id
+                for id in ids
+                if get_safe(id, "external-id-relationship", "")
+                in ["self", "version-of", "part-of"]
```
Comment (falquaddoomi) [inline _cite/plugins/orcid.py:50]: Based on https://support.orcid.org/hc/en-us/articles/360006894774-Group-multiple-versions-of-the-same-work-together I think that you should always choose "self" over "version-of"  or "part-of". I'm unsure if the list is always sorted with "self" coming before the others. In the few examples I could find (e.g., [Casey's publication history](https://pub.orcid.org/v3.0/0000-0001-8713-9213/works)) that seemed to be the case, in which case you don't need to change this.

To be sure, though, you might consider sorting `ids` according to the relationship priority, e.g.:

```python
# little helper used for sorting
# gets the position of "v" in iterable "l" or puts it at the end if not found
def index_or(l, v):
  return l.index(v) if v in l else len(l)
  
# to get ids sorted in the order specified in the list, with anything not recognized coming after
sorted(ids, key=lambda x: index_or(["self", "part-of"], get_safe(id, "external-id-relationship", "")))

# you'd use that sorted() expression above as the thing you're iterating over
```

It's not the world's most efficient code, but it should be fine for the small number of elements you're dealing with.

**Review by falquaddoomi — COMMENTED** [review]
Sorry for the delay; I made some comments on the Python code. All the other changes seemed straightforward to me.

File: _cite/plugins/orcid.py
[the remark this answers is not in the document]
Comment (vincerubinetti) [reply _cite/plugins/orcid.py:1]: I think you're right, though based on those docs and your test, I think the work summaries should already be sorted in `display-index` order, so that block of code isn't needed at all.

> The display index also determines the work order when reading the ORCID record.

[the remark this answers is not in the document]
Comment (vincerubinetti) [reply _cite/plugins/orcid.py:1]: Screenshot 2024-05-30 at 12 59 40 PM

File: _cite/plugins/orcid.py (hunk 1 above)
Comment (vincerubinetti) [reply _cite/plugins/orcid.py:50]: For Casey's ORCID I'm actually not seeing any ID that's not "self".

<details><summary>script</summary>

```js
const response = await (
  await fetch("https://pub.orcid.org/v3.0/0000-0001-8713-9213/works", {
    headers: { Accept: "application/json" },
  })
).json();
new Set(
  response.group
    .map((g) =>
      g["external-ids"]["external-id"].map(
        (id) => id["external-id-relationship"]
      )
    )
    .flat()
);
```

</details>

I feel like maybe we should keep the "order is the priority" approach here that we're now doing for the work summaries, even though I don't see any documentation to support it.

[This doc page](https://support.orcid.org/hc/en-us/articles/360006971353-Metadata-in-the-Works-section?input_string=api+help+with+%2Fworks+endpoint) explains a bit more what the relationship types could be. Notably, preprints can show up as "version of", which is why I wanted to include it as well. Conceivably, an author could specify that the preprint is the preferred source (probably temporarily) and it might show up first in the list.

Again, wish I had someone from ORCID to consult with. Still haven't been approved access to that google group for help. A lot of this could come down to user preference as well, so keeping it easy to tweak the script is also a priority here.

**Comment by vincerubinetti** [comment]
@andrewsu Do you have time to give this code a quick look and these set of issues one last think?

File: _cite/plugins/orcid.py (hunk 1 above)
Comment (vincerubinetti) [reply _cite/plugins/orcid.py:50]: Okay that settles it for me, for now. Let's keep it as is. Can always revert the behavior in a future version... and in lieu of any ORCID authority, seems like the only way to truly find out if it works is to just try it out with many users and see what happens.

Comment (vincerubinetti) [reply _cite/plugins/orcid.py:49]: @falquaddoomi Regarding your idea of having the plugin here match what the ORCID website UI does. I think that's a good idea, though for this part of the code in particular, I don't see anything in their UI that matches this data. They only seem to have a [search](https://orcid.org/orcid-search/search) for authors, not individual papers. And looking at a particular [author page](https://orcid.org/0000-0003-4297-8747), I don't see any UI elements with an order other than the work-summary user-specified order that we've been discussing.

Also related... I'm going to remove the line of code that includes the top level `external-ids` after the `work-summary` `external-ids`. Testing the following script with Casey's, Andrew's, and Faisal's ORCIDs, the top level ids are always a subset of the work summary ids (sometimes the latter has one or two extra). I guess they just duplicate the ids at the top level for convenience (though the fact that they're not exactly the same as the work summary ids is bound to cause issues).

<details><summary>script</summary>

```js
const result = await (
  await fetch("https://pub.orcid.org/v3.0/0000-0001-8713-9213/works", {
    headers: { Accept: "application/json" },
  })
).json();

for (const entry of result.group) {
  const ids = new Set(
    entry["external-ids"]["external-id"].map((id) => id["external-id-value"])
  );
  const summaries = new Set(
    entry["work-summary"]
      .map((summary) =>
        summary["external-ids"]["external-id"].map(
          (id) => id["external-id-value"]
        )
      )
      .flat()
  );
  if (!ids.isSubsetOf(summaries)) console.log({ ids, summaries });
}
```

</details>

IN CONCLUSION... what we're left with is dead simple and I think the most correct. Go through work summaries in order (user preference), and take the first available one (as long as it has an acceptable relationship type). I thought about adding code to sort manubot-citable ID types to the front of the list (in case the first entry is a [weird/obscure type](https://github.com/manubot/manubot/issues/365)). But then I thought it's probably better to respect the user-selected preference. They may have a good reason for putting that weird ID first. The plugin will keep the ORCID citation details anyway.

Hopefully this settles the matter.
